### DISCUSSION TITLE
Pocketbase hosting on fly.io keeps running out of memory

### DISCUSSION THREAD

**Discussion by lukasm03** [pocketbase/pocketbase#1359] [Q&A]
I have played around with pocketbase for a side project and I really like it so far! But I have this issue where the instance keeps crashing due to memory issues. The error message is following:
`[ 49.641019] Out of memory: Killed process 528 (pocketbase) total-vm:1238208kB, anon-rss:201512kB, file-rss:0kB, shmem-rss:0kB, UID:0 pgtables:900kB oom_score_adj:0`

The error happens when I open the main page, which loads in about 100 images with corresponding details of where the image was taken and date, and so on. Is this an issue with pocketbase, and if not, does anybody know how to fix this?

**Comment by ganigeorgiev** [comment]
What version of PocketBase are you using?

Several performance and memory improvements were done in the latest v0.10+ releases (see the [CHANGELOG](https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md) for details).

If you are using PocketBase v0.10+ and still are getting OOM errors, you can try defining `GOMEMLIMIT=200MiB` env variable.

**Reply by lukasm03** [reply]
I am using the version 0.10.1, will try your fix and provide an update.

**Reply by ganigeorgiev** [reply]
Alternatively, you can also try disabling the logs by setting "Admin UI > Application > Logs max days retention" to `0`.

It is also possible that the OOM errors are related to the thumb generations (if you are using custom size(s) in addition to the default 100x100).

_Also please note that if you are on the free fly.io plan, it offers very little memory (256MB) and it is suitable mostly for testing/small applications, so depending on your app load and what and how many requests you are sending you may need to upgrade._

**Reply by ganigeorgiev** [reply]
@lukasm03 You can also check this discussion how to enable swap to help with the OOM errors - https://community.fly.io/t/swap-memory/2749.
